src/hugr/validate.rs

```diff
@@ -558,6 +528,62 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
         }
         .into())
     }
+
+    fn validate_subtree(&self, node: Node, var_decls: &[TypeParam]) -> Result<(), ValidationError> {
+        let op_type = self.hugr.get_optype(node);
+        // TODO consider turning this match into a trait method?
+        match op_type {
+            OpType::LeafOp(crate::ops::LeafOp::CustomOp(b)) => {
+                // Try to resolve serialized names to actual OpDefs in Extensions.
+                let e: Option<ExtensionOp>;
+                let ext_or_opaq = match &**b {
```

Comment:
This was quite hard to get right, but the Either is necessary because we want only a *reference* - an `&ExtensionOp` or `&OpaqueOp`, whereas an `ExternalOp` stores an actual `ExtensionOp` or `OpaqueOp`

Reply:
No, managed to do this with ExternalOp after all